src/components/WeatherList/WeatherList.tsx

```diff
@@ -0,0 +1,117 @@
+import { useContext, useState } from 'react';
+import { format } from 'date-fns';
+import Collapse from '@mui/material/Collapse';
+import List from '@mui/material/List';
+import ListItem from '@mui/material/ListItem';
+import ListItemButton from '@mui/material/ListItemButton';
+import ListItemText from '@mui/material/ListItemText';
+import ListSubheader from '@mui/material/ListSubheader';
+import Tooltip from '@mui/material/Tooltip';
+import SettingsContext from 'contexts/SettingsContext';
+import { Forecast } from 'types/forecast';
+import Location from 'types/location';
+import SettingsContextType from 'types/settingsContextType';
+import findCountryNameByCode from 'utils/findCountryNameByCode';
+import formatTemperatureData from 'utils/formatTemperature';
+import * as classes from './styles';
+
+interface Props {
+  forecast: Forecast;
+  location: Location;
+  className?: string;
+}
+
+export default function WeatherList({ forecast, location, className }: Props) {
+  const { selectedTemperatureUnit } =
+    useContext<SettingsContextType>(SettingsContext);
+  const [openItems, setOpenItems] = useState<boolean[]>(
```

Comment:
Do you need this state at all?

Reply:
Yes, because I use it to control which list item should be collapsed or expanded. But maybe there is a better way to do this :)